src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1758,94 +1748,92 @@ private bool IsInTypeofExpression(int position)
+                    resultKind = highestResultKind;
+                    isDynamic = highestIsDynamic;
+                }
+                else if (highestResultKind != LookupResultKind.Empty && highestResultKind < resultKind)
+                {
+                    resultKind = highestResultKind;
+                    isDynamic = highestIsDynamic;
+                }
+                else if (highestBoundExpr.Kind == BoundKind.TypeOrValueExpression)
+                {
+                    symbols = highestSymbols;
+                    resultKind = highestResultKind;
+                    isDynamic = highestIsDynamic;
+                }
+                else if (highestBoundExpr.Kind == BoundKind.UnaryOperator)
+                {
+                    if (IsUserDefinedTrueOrFalse((BoundUnaryOperator)highestBoundExpr))
                     {
                         symbols = highestSymbols;
                         resultKind = highestResultKind;
                         isDynamic = highestIsDynamic;
                     }
-                    else if (highestBoundExpr.Kind == BoundKind.UnaryOperator)
+                    else
                     {
-                        if (IsUserDefinedTrueOrFalse((BoundUnaryOperator)highestBoundExpr))
-                        {
-                            symbols = highestSymbols;
-                            resultKind = highestResultKind;
-                            isDynamic = highestIsDynamic;
-                        }
-                        else
-                        {
-                            Debug.Assert(ReferenceEquals(lowestBoundNode, highestBoundNode), "How is it that this operator has the same syntax node as its operand?");
-                        }
+                        Debug.Assert(ReferenceEquals(lowestBoundNode, highestBoundNode), "How is it that this operator has the same syntax node as its operand?");
                     }
                 }
+            }
 
-                if (resultKind == LookupResultKind.Empty)
-                {
-                    // Empty typically indicates an error symbol that was created because no real
-                    // symbol actually existed.
-                    return SymbolInfoFactory.Create(ImmutableArray<Symbol>.Empty, LookupResultKind.Empty, isDynamic);
-                }
-                else
+            if (resultKind == LookupResultKind.Empty)
+            {
+                // Empty typically indicates an error symbol that was created because no real
+                // symbol actually existed.
+                return SymbolInfoFactory.Create(ImmutableArray<Symbol>.Empty, LookupResultKind.Empty, isDynamic);
+            }
+            else
+            {
+                // Caas clients don't want ErrorTypeSymbol in the symbols, but the best guess
+                // instead. If no best guess, then nothing is returned.
+                var builder = ArrayBuilder<Symbol>.GetInstance();
+                foreach (var s in symbols)
```

Comment:
FYI.  Compiler style is to be explicit about the type *unless* it is clear from the initializer.  So in this case, keep as ```Symbol symbol```